chains/ethereum/listener.go

```diff
@@ -152,6 +152,7 @@ func (l *listener) getDepositEventsForBlock(latestBlock *big.Int) error {
 			m, err = l.handleGenericDepositedEvent(destId, nonce)
 		} else {
 			l.log.Error("Event has unrecognized handler", "handler", addr)
+			return nil
```

Comment:
Please return an error here. You can duplciate whats in the log if you want

Reply:
Sounds good
```
l.log.Error("handler", addr)
return errors.New("Event has unrecognized handler")
```
I realized I could have just concatenated the error text and `addr` into one string, so I don't log an error and also return one.

Reply:
Generally it is a bad idea, you'll end up with a bunch of useless duplicate logs. 

I don't actually see any reason to make an exception here, the caller will log it, so we only need to return it :+1: